### ISSUE TITLE
GdipGetLineColors has different call signatures for 32-bit versus 64-bit

### ISSUE THREAD

**Issue by GCuser99** [fafalone/WinDevLib#18]
I've confirmed that the 32- and 64-bit versions have different call signatures by using both in tB (and of course they are not interchangeable). The 32-bit version requires a Long array of dimension 2 (Colors(0) and Colors(1)).

```vba
#If Win64 Then
Private Declare PtrSafe Function GdipGetLineColors Lib "gdiplus" (ByVal brush As LongPtr, color1 As Long, color2 As Long) As GpStatus
#Else
Private Declare Function GdipGetLineColors Lib "gdiPlus" (ByVal brush As Long, Colors As Any) As GpStatus
#End If
```

Thx!

**Comment by fafalone** [comment]
I see how the current definition is wrong, but not how that would be right? gdiplusflat.h:

```C
GpStatus WINGDIPAPI
GdipGetLineColors(GpLineGradient *brush, ARGB* colors);

//gdipluspixelformats.h
typedef DWORD ARGB;
```

So the definition should be the same in both, `    Public Declare PtrSafe Function GdipGetLineColors Lib "gdiplus" (ByVal brush As LongPtr, colors As Long) As GpStatus` because a `DWORD` just a 4-byte `Long` on both 32bit and 64bit.

**Comment by GCuser99** [comment]
Oh, that's strange! You are right that:
```vba
Public Declare PtrSafe Function GdipGetLineColors Lib "gdiplus" (ByVal brush As LongPtr, colors As Long) As GpStatus
```
...works for both!

But the two-color argument version does "seem" to work for 64-bit and not 32-bit! Here is a snippet:
```vba
Dim Values(0 To 1) As Long
If GdipGetLineColors(BrushHandle, Values(0), Values(1)) = 0& Then
```
Maybe the second argument is superfluous and gets ignored?

Thanks!

**Comment by GCuser99** [comment]
Yeah, that is what it is... here is a test in tB which fails in 32-bit but works in 64-bit. Not sure what the expected behavior is (don't have VB6 to test).
```vba
Private Declare PtrSafe Sub CopyMemory Lib "kernel32.dll" Alias "RtlMoveMemory" ( _ 
    ByRef Destination As Any, _
    ByRef Source As Any, ByVal Length As Long, _
    ByVal superfluous As Long) '<-- added an unsupported argument

Sub test()
    Dim a As Long, b As Long
    a = 1: b = 2
    CopyMemory a, b, 4, 99999 '<-- fails in 32-bit tB with "Bad DLL Definition"
    Debug.Print a 'prints 2 in 64-bit tB
End Sub
```

**Comment by fafalone** [comment]
Well as long as the new version works in both 32bit and 64bit, I'll leave why the superfluous parameter works in 64bit as a mystery for those more familiar with the calling conventions on a low level.

Fixed in v7.8.379, available now.

Sorry for the multiple GDIP bugs; thanks for helping make WinDevLib better by reporting them though 👍
